### DISCUSSION TITLE
integrate new categories for sensors

### DISCUSSION THREAD

**Discussion by maxgeilert** [jseidl/magic-areas#185] [Ideas / Feature Requests]
hello,
I'd love to use magic areas in combination with room-assistant for a solid presence detection.
It would be nice to be able to have certain sensors be able to turn the area to occupied while others are only allowed to keep an area occupied when it already is.
I used motion sensors as an inital trigger (for now triggering an timer) and allowed room-assistant to restart the timer only when it is already running. The problem with room-assistant is the fluctuation between rooms. I installed magic areas and had to create a template sensor for each room because I couldn't figure out a way to integrate the room sensor directly. Since the bluetooth tracking jumps from room to room sometimes, those binary_sensor templates would turn on instantly and magic areas would turn on the light in rooms noone is in.

**Comment by jseidl** [comment]
Interesting problem... we have some users using room-assistant with binary sensors but I'd never though of this fluctuation problem. Added to our backlog to investigate this solution, thanks for the report

**Comment by jseidl** [comment]
Can you share the template sensor you used to integrate with room assistant?

**Comment by maxgeilert** [comment]
Sorry for the delay, I didn't get a notification..just checked, I deleted those templates a while back, but this is more or less what I did for every member of the household and every room:
`max_living:
        friendly_name: "max livingroom"
        value_template: >-
          {{ states('sensor.max_room_presence') }}
          {% if is_state('sensor.max_room_presence', 'livingroom') %}
          on
          {%- else -%}
          off
          {%- endif %}`

**Comment by jseidl** [comment]
Hey Max! Sorry it took so long to circle back on this. My daughter was born later that month (when you replied) and life's been crazy as you may expect.

I believe a solution for not having to manually write templates for every room would be to track a given sensor and consider it when its value is the same as the area name / area id. The tricky part is that there isn't a sensor device class that would fit that (say `presence`, as we have for binary sensors). I need to think a bit more on how to implement this without needing too much user configuration (otherwise isn't magic!)

For the "debounce" issue, yeah I can see a field for listing which sensors you'd like to use just to keep state but not to trigger. A workaround would be using `delay_on` https://www.home-assistant.io/integrations/template/#delay_on on the templates, which would also kinda make them "keep only" sensors.

**Comment by jseidl** [comment]
I think this was solved already by the "keep only sensors" and "BLE tracker sensor" features, please reopen if I'm wrong
